lib/internal/http2/compat.js

```diff
@@ -181,6 +181,10 @@ class Http2ServerRequest extends Readable {
     return '2.0';
   }
 
+  get socket() {
+    return this.stream.session.socket;
+  }
```

Comment:
Oops, sorry for sneaking it into this PR. 🙈

In this case I'm using it to report the `alpnProtocol` from the underlying TLS connection. I think there could be other valid use cases. $0.02

If we explicitly *do not* implement this part of the compatibility layer, maybe it should throw or emit a warning?